src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlProcessingInstruction.cs

```diff
@@ -14,94 +14,93 @@ public class XmlProcessingInstruction : XmlLinkedNode
         private readonly string _target;
         private string _data;
 
-        protected internal XmlProcessingInstruction(string target, string data, XmlDocument doc) : base(doc)
+        protected internal XmlProcessingInstruction(string target, string? data, XmlDocument doc) : base(doc)
         {
-            _target = target;
-            _data = data;
-        }
+            Debug.Assert(target != null, "All callsites should have already checked for null");
 
-        // Gets the name of the node.
-        public override string Name
-        {
-            get
+            if (string.IsNullOrEmpty(target))
             {
-                if (_target != null)
-                    return _target;
-                return string.Empty;
+                throw new ArgumentException(SR.Xml_EmptyName);
```

Comment:
Shouldn't this also include the argument name as the second argument?

Reply:
Also, we generally will instead do something like:
```C#
ArgumentNullException.ThrowIfNull(target);
if (target.Length == 0)
{
    throw new ArgumentException(SR.Xml_EmptyName, nameof(target));
}
```
so that we throw ANE when it's null.

Reply:
we do throw `Xml_EmptyName` but I believe it's because it's being parsed and therefore there is no specific argument. I'll add new resource string

Reply:
Actually we do throw similar errors elsewhere as well, i.e. XmlDocument.CreateElement("") will throw:

```
Unhandled exception. System.ArgumentException: The local name for elements or attributes cannot be null or an empty string.
```

Reply:
You don't want to at least include `, nameof(target)` in the exception?

Reply:
🤦 that's what happens when you update PR late at night, I forgot ArgumentException has that argument and for some reason though you suggested to add placeholder to the resource string. Will change 😄